frame/multisig/src/lib.rs

```diff
@@ -425,6 +434,17 @@ decl_module! {
 				ensure!(m.when == timepoint, Error::<T>::WrongTimepoint);
 				ensure!(m.approvals.len() < threshold as usize, Error::<T>::NoApprovalsNeeded);
 				if let Err(pos) = m.approvals.binary_search(&who) {
+					if m.approvals.len() + 1 == threshold as usize {
+						if let Some(call) = Self::take_call(&call_hash) {
+							let result = call.dispatch(frame_system::RawOrigin::Signed(id.clone()).into());
```

Comment:
This also should change the weight. You are running into the same issues I had in collective which also has stored calls that are (potentially) executed later.

Reply:
See `weight_bound` param [here](https://github.com/paritytech/substrate/blob/606c56d2e2f69f68f3947551224be6a3515dff60/frame/collective/src/lib.rs#L647) and the `validate_and_get_proposal` function [here](https://github.com/paritytech/substrate/blob/606c56d2e2f69f68f3947551224be6a3515dff60/frame/collective/src/lib.rs#L756).
We might want to build some utilities around this use case of deferred calls so it's not as much effort to make it secure weight-wise every time.